templates/common/_base/files/configure-ovs-network.yaml

```diff
@@ -0,0 +1,116 @@
+mode: 0755
+path: "/usr/local/bin/configure-ovs.sh"
+contents:
+  inline: |
+    #!/bin/bash
+    set -eux
+
+    # Configures NICs onto OVS bridge "br-ex"
+    # Configuration is either auto-detected or provided through a config file written already in Network Manager
+    # key files under /etc/NetworkManager/system-connections/
+    # Managing key files is outside of the scope of this script
+
+    iface=""
+    counter=0
+    # find default interface
+    while [ $counter -lt 12 ]; do
+      iface=$(ip -j route show default | jq -r '.[0].dev')
```

Comment:
Isn't the loop unnecessary since this is `After=NetworkManager-wait-online.service`? It should be guaranteed that a default route already exists.

Reply:
I don't think so because its a Wants+After and not a Requires. Therefore if the unit fails during start up (never gets online) this unit will still run.

Reply:
but in that case, waiting another 60 seconds isn't going to help; we can just run `ip route show default` once, and if it fails, exit immediately

Reply:
well it might, because NM will still be trying to DHCP that interface even though wait online has already failed. But we can take this to a follow up.